src/features/network-modification-table/utils.ts

```
@@ -22,6 +22,18 @@ export function isCompositeModification(modification: ComposedModificationMetada
     return modification?.messageType === MODIFICATION_TYPES.COMPOSITE_MODIFICATION.type;
 }
 
+// returns the depth of the modification with the given uuid in the given mods tree
+export function findDepth(mods: ComposedModificationMetadata[], uuid: UUID, currentDepth = 0): number {
+    for (const mod of mods) {
+        if (mod.uuid === uuid) return currentDepth;
+        if (mod.subModifications?.length) {
+            const found = findDepth(mod.subModifications, uuid, currentDepth + 1);
+            if (found >= 0) return found;
+        }
+    }
```

ESLint

[error] iterators/generators require regenerator-runtime, which is too heavyweight for this guide to allow them. Separately, loops should be avoided in favor of array iterations.

(no-restricted-syntax)